spec/controllers/passengers_controller_spec.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe PassengersController do
+
+  let!(:admin) { create :user, :admin }
+  before :each do
+    session[:user_id] = admin.id
+  end
+
+  describe 'POST set_status' do
+    context 'with broken mailer' do
+      context 'changing active to archived' do
+        it 'saves the passenger and displays a warning message' do
+          allow(PassengerMailer).to receive(:notify_archived).and_raise Net::SMTPFatalError
```

Comment:
I would move this to a `before` block inside `'with a broken mailer'`. It's setup that defines that context.